src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -26,12 +73,17 @@ const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucke
     }
   }
 
+  const isDeleteDisabledWithApps = (deletableApps, nonDeletableApps) => hasApps(deletableApps, nonDeletableApps) && nonDeletableApps.length !== 0
+
   return h(Modal, {
     title: 'Delete workspace',
     onDismiss,
     okButton: h(ButtonPrimary, {
-      disabled: _.toLower(deleteConfirmation) !== 'delete workspace',
-      onClick: () => deleteWorkspace()
+      disabled: _.toLower(deleteConfirmation) !== 'delete workspace' || isDeleteDisabledWithApps(deletableApps, nonDeletableApps),
+      onClick: () => deleteWorkspace(),
+      tooltip: _.toLower(deleteConfirmation) !== 'delete workspace' ?
+        isDeleteDisabledWithApps(deletableApps, nonDeletableApps) ? 'You must ensure all apps in this workspace are deletable' : 'You must type the confirmation message' :
```

Comment:
this nested conditional gives us the 3 different contextual tooltips shown in screenshots